internal/orchestration/components.go

```diff
@@ -121,24 +121,156 @@ func getCrypto(
 	return
 }
 
-type moduleList struct {
+type CoreComponents struct {
+	Options   *core.Options
+	EventLoop *eventloop.EventLoop
+	Logger    logging.Logger
+}
+
+func NewCoreComponents(
+	id hotstuff.ID,
+	logTag string,
+	privKey hotstuff.PrivateKey,
+) *CoreComponents {
+	opts := core.NewOptions(id, privKey)
```

Comment:
Stuff like this can be done directly:
```go
return &CoreComponents{
	Options:   core.NewOptions(id, privKey),
	Logger: logging.New(fmt.Sprintf("%s%d", logTag, id)),
	EventLoop: eventloop.New(logger, 100),
}
```